ast/src/expression/bind.rs

```diff
@@ -115,14 +117,14 @@ mod tests {
     fn test() {
         let e = Bind::Let {
             identifier: "x".to_string(),
-            value: Box::new(Primitive::UInt64(5)),
-            body: Box::new(Apply(
-                Box::new(Apply(
-                    Box::new(Binary::Equals),
-                    Box::new(Primitive::UInt64(5)),
-                )),
-                Box::new(Rvalue(Var::Bind("x".to_string()))),
-            )),
+            value: Box::new(Expr::Primitive(Primitive::UInt64(5))),
+            body: Box::new(Expr::Apply(Apply(
+                Box::new(Expr::Apply(Apply(
```

Comment:
although i agree with the design decisions necessitating this verbosity, i hate it. imma write some expression generator macros